src/base/bittorrent/portforwarderimpl.cpp

```diff
@@ -55,61 +53,71 @@ bool PortForwarderImpl::isEnabled() const
 
 void PortForwarderImpl::setEnabled(const bool enabled)
 {
-    if (m_storeActive != enabled)
-    {
-        if (enabled)
-            start();
-        else
-            stop();
+    if (m_storeActive == enabled)
+        return;
 
-        m_storeActive = enabled;
-    }
+    if (enabled)
+        start();
+    else
+        stop();
+    m_storeActive = enabled;
 }
 
 void PortForwarderImpl::addPort(const quint16 port)
 {
-    if (!m_mappedPorts.contains(port))
-    {
-        m_mappedPorts.insert(port, {});
-        if (isEnabled())
-            m_mappedPorts[port] = {m_provider->add_port_mapping(lt::session::tcp, port, port)};
-    }
+    if (m_mappedPorts.contains(port))
+        return;
+
+    const std::vector<lt::port_mapping_t> mapping = isEnabled()
```

Comment:
IMO, too many internals for no reason. Maybe:
```c++
if (isEnabled())
    m_mappedPorts.insert(port, m_provider->add_port_mapping(lt::session::tcp, port, port));
else
    m_mappedPorts.insert(port, {});
```

Reply:
All comment addressed.